core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -607,7 +595,7 @@ public void deleteShardStore(String reason, ShardId shardId, ClusterState cluste
      * @return true if the index can be deleted on this node
      */
     public boolean canDeleteIndexContents(Index index, IndexSettings indexSettings, boolean closed) {
-        final IndexService indexService = this.indices.get(index.getName());
+        final IndexService indexService = indexService(index);
```

Comment:
same comment about checking for index with the same name, which probably means we need a synchronized close so the map won't change?